File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int start=0;
        int end=height.length-1;
        int maxArea=0;
        while(start<end){
            int h=Math.min(height[start],height[end]);
            int w=end-start;
            int area=h*w;
            maxArea=Math.max(area,maxArea);
            if(height[start]<height[end]){
                start++;        
            }else{
                end--;
            }
        } 
        return maxArea;   
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        start = 0
        end = len(height)-1
        maxWater = 0
        while start < end:
            b = end - start
            h = min(height[start],height[end])
            maxWater = max(maxWater,h*b)
            if height[start] < height[end]:
                start +=1
            else:
                end -=1    
        return maxWater 





        